crates/sovereign-sdk/adapters/risc0-bonsai/src/host.rs

```diff
@@ -383,7 +390,14 @@ impl<'a> ZkvmHost for Risc0BonsaiHost<'a> {
             tracing::info!("SNARK session created: {}", snark_session.uuid);
 
             loop {
-                let res = client.snark_status(&snark_session)?;
+                let res = match client.snark_status(&snark_session) {
+                    Ok(res) => res,
+                    Err(_) => {
+                        tracing::warn!("Failed to get status, retrying...");
+                        std::thread::sleep(Duration::from_secs(15));
+                        continue;
+                    }
+                };
```

Comment:
from what I understand error at that point will either be a connection error (can't get response from bonsai skd) or it will be an internal error from the API. so waiting here makes sense imo.

other type of errors, let's say session was not created but we ask for status, will fall in the _ arm of the match